File: Loader/src/main/java/me/kyroclient/transformer/SafeTransformer.java
package me.kyroclient.transformer;

import java.lang.instrument.ClassFileTransformer;
import java.lang.instrument.IllegalClassFormatException;
import java.security.ProtectionDomain;

public abstract class SafeTransformer implements ClassFileTransformer {
    public abstract byte[] transform(ClassLoader loader, String className, byte[] originalClass);
    @Override
    public byte[] transform(ClassLoader loader, String className, Class<?> classBeingRedefined, ProtectionDomain protectionDomain, byte[] classfileBuffer) throws IllegalClassFormatException {
        return this.transform(loader, className, classfileBuffer);
    }
}

File: Loader/src/main/kotlin/me/kyroclient/transformer/SafeTransformer.kt
package me.kyroclient.transformer

import java.lang.instrument.ClassFileTransformer
import java.security.ProtectionDomain

internal interface SafeTransformer : ClassFileTransformer {

    fun transform(loader: ClassLoader, className: String, originalClass: ByteArray): ByteArray?

    override fun transform(
        loader: ClassLoader,
        className: String,
        classBeingRedefined: Class<*>?,
        protectionDomain: ProtectionDomain?,
        classfileBuffer: ByteArray
    ):ByteArray? {
        try {
            return this.transform(loader, className, classfileBuffer)
        } catch (e: Throwable) {
            e.printStackTrace()
            System.exit(1)
        }

        return null;
    }

}